src/cli/templates/decorators/with-page-metadata.ts

```diff
@@ -28,7 +28,7 @@ export const tplDynamicForDynamicRoute = `
 import {generateMetadata as generateMetadataOrigin} from '${PATTERNS.originPath}'
 
 export async function generateMetadata({ params, ...otherProps }:any) {
-  return generateMetadataOrigin({ ...otherProps, params, pageHref: compileHref('${PATTERNS.pageHref}', params) })
+  return generateMetadataOrigin({ ...otherProps, params, pageHref: compileHref('${PATTERNS.pageHref}', await params) })
```

Comment:
How about passing "getPageHref" callback and omit awaiting params before needed?

Reply:
Sounds good! I will also add passing the locale to these generateMetadataOrigin and generateViewportOrigin functions so that the locale can be used synchronously.

Reply:
Changes done

Reply:
Looks good, thank you.